diff --git a/modules/migration_mapper/src/Plugin/DataSource/Json.php b/modules/migration_mapper/src/Plugin/DataSource/Json.php
index 68f6160..21b2be2 100644
--- a/modules/migration_mapper/src/Plugin/DataSource/Json.php
+++ b/modules/migration_mapper/src/Plugin/DataSource/Json.php
@@ -38,8 +38,11 @@ class Json extends DataSourceBase {
 
     if (!empty($this->sourcePaths)) {
       foreach ($this->sourcePaths as $key => $value) {
-        if (strpos($key, $config['source']['json_item_selector']) !== FALSE) {
-          $val = str_replace($config['source']['json_item_selector'], '', $key);
+        $item_selector = $config['source']['json_item_selector'];
+        // If item selector has is not have '/' in last, we add that.
+        $item_selector = substr($item_selector, -1) != '/' ? $item_selector . '/' : $item_selector;
+        if (strpos($key, $item_selector) !== FALSE) {
+          $val = str_replace($item_selector, '', $key);
           unset($this->sourcePaths[$key]);
           if (empty($val)) {
             continue;
